app/models/spree/image_decorator.rb

```diff
@@ -14,7 +14,7 @@ def update_mailchimp_product
       end
 
       def find_product
-        Spree::Variant.find(self.viewable_id).product.id
+        Spree::Variant.find(viewable_id).product.id
```

Comment:
`Spree::Variant.unscoped.find(viewable_id).product_id`

1. One SQL query less
2. It will also return removed/sold out Variants